source/faq.txt

```diff
@@ -4,12 +4,176 @@ FAQ
+   * - ``mongo-java-driver``
+     - An uber jar containing the bson library, the core library and the mongodb-driver. This artifact is a valid OSGi bundle.
+   * - ``mongodb-driver-core``
+     - The core library, (new in 3.0). **No BSON included (but is a dependency), and no high-level drivers.**
+
+POJOs
+-----
+
+Do I have to specify an ID field value myself?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Imagine you have the following entity:
+
+.. code-block:: java
+
+   public class Entity {
+      public final ObjectId id;
+      public String value;
+
+      public Entity(){ }
+   }
+
+When inserting with ``collection.insertOne()``, the Codec will encode
+this into JSON, but skip the ``_id`` field since the value is ``null``.
+However, the server automatically generates an ``_id`` for the document
+during the insert, but the value of this ``_id`` is not passed back via
+the driver. 
+
+.. note::
+
+   You can mark any one field to encode as ``_id`` with the @BsonId
+   annotation. 
+   
+   With the entity listed above, you can mark the ``id`` field to encode
+   as ``_id``:
+
+   .. code-block:: java
+
+      @BsonId()
+      public final ObjectId id;
+
+Can the ID field be a compound key?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Yes. For an example of this, see `our implementation <https://github.com/niccottrell/mongo-java-tests/blob/master/src/test/PojoCompoundIdTest.java>`_
+
+What is the discriminator?
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The discriminator is for cases where you want to use inheritance, and
+store multiple types of documents within the same collection or parent
+document (in case you embed sub-documents). 
+
+For example, if you have an ``Event`` class, that you extend in Java (e.g.
+``MachineEvent`` or ``NetworkEvent``), using the discriminator identifies
+which class the PojoCodec should use to serialize/deserialize the
+document. 
+
+Can I control serialization of ``LocalDate``?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Yes, the 3.7 Java driver adds native support for ``JSR-310 Instant``,
+``LocalDate`` & ``LocalDateTime``. 
+
+Add the Codec to the top your registry before the ``DefaultCodecRegister``
+and before the ``PojoCodecProvider``: 
+
+.. code-block:: java
+
+   static final CodecRegistry CODEC_REGISTRY = fromRegistries(
+      CodecRegistries.fromCodecs(
+                        new MyDateAsStringCodec()),
+                        MongoClient.getDefaultCodecRegistry(),
+                        fromProviders(pojoCodecProvider));
+
+Can I serialize a ``java.util.Date`` as a string in format **yyyy-mm-dd**?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Yes, you can build your own Codec for this class and add it to the registry. 
+
+Can I make POJOs read/write directly to the field and not use the getters/setters at all?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Not natively, but you can build your own POJO Codec.
```

Comment:
I would remove this line and just say you can configure the PojoCodecProvider to use the 
``SET_PRIVATE_FIELDS_CONVENTION`` to set a private field via reflection if no public setter is available.